xray发起udp崩溃

### 完整性要求

- [X] 我保证阅读了文档，了解所有我编写的配置文件项的含义，而不是大量堆砌看似有用的选项或默认值。
- [X] 我提供了完整的配置文件和日志，而不是出于自己的判断只给出截取的部分。
- [X] 我搜索了 issues, 没有发现已提出的类似问题。
- [X] 问题在 Release 最新的版本上可以成功复现

### 描述

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x6b510b]
goroutine 63 [running]:
github.com/xtls/xray-core/common/net.Destination.NetAddr({{0x0?, 0x0?}, 0x
ee08?, 0xc0?})
        github.com/xtls/xray-core/common/net/destination.go:93 +0x2b
github.com/xtls/xray-core/common/net.Destination.String({{0x0?, 0x0?}, 0x0
?, 0x0?})
        github.com/xtls/xray-core/common/net/destination.go:140 +0x74
github.com/xtls/xray-core/common/serial.ToString({0x12b53e0?, 0xc0004bc408
?})
        github.com/xtls/xray-core/common/serial/string.go:20 +0x12a
github.com/xtls/xray-core/common/log.(*AccessMessage).String(0xc00032f800)
        github.com/xtls/xray-core/common/log/access.go:36 +0x154
github.com/xtls/xray-core/common/log.(*generalLogger).run(0xc0000816c0)
        github.com/xtls/xray-core/common/log/logger.go:89 +0x1ea
created by github.com/xtls/xray-core/common/log.(*generalLogger).Handle in
 goroutine 58
        github.com/xtls/xray-core/common/log/logger.go:109 +0x8f

### 重现方式

发送任意UDP数据包即可复现(例如使用[udpping:https://github.com/wangyu-/UDPping]或是使用speed.cloudflare.com测速)

### 客户端配置

<details><pre><code>
N/A
</code></pre></details>

### 服务端配置

<details><pre><code>
{
  "log": {
    "loglevel": "debug",
    "access": "/root/xray/access.log",
    "error": "/root/xray/error.log"
  },
  "inbounds": [
    {
      "listen": "@xhttp",
      "protocol": "vless",
      "settings": {
        "decryption": "none",
        "clients": [
          {
            "id": "xxxx"
          }
        ]
      },
      "streamSettings": {
        "network": "xhttp",
        "xhttpSettings": {
          "path": "/yourpath"
        }
      }
    },
    {
      "listen": "0.0.0.0",
      "port": 443,
      "protocol": "vless",
      "settings": {
        "decryption": "none",
        "fallbacks": [
          {
            "dest": "@xhttp"
          }
        ]
      },
      "streamSettings": {
        "network": "raw",
        "security": "reality",
        "realitySettings": {
          "target": "xxxx.xxx:443",
          "serverNames": [
            "xxxx.xxx"
          ],
          "privateKey": "xxxx",
          "shortIds": [
            ""
          ]
        }
      }
    },
    {
      "listen": "0.0.0.0",
      "port": 8443,
      "protocol": "vless",
      "settings": {
        "decryption": "none",
        "fallbacks": [
          {
            "dest": "@xhttp"
          }
        ]
      },
      "streamSettings": {
        "network": "raw",
        "security": "reality",
        "realitySettings": {
          "target": "xxxx.xxx:443",
          "serverNames": [
            "xxxx.xxx"
          ],
          "privateKey": "xxxx",
          "shortIds": [
            ""
          ]
        }
      }
    }
  ],
  "outbounds": [
    {
      "tag": "direct",
      "protocol": "freedom"
    }
  ]
}
</code></pre></details>

### 客户端日志

<details><pre><code>
N/A
</code></pre></details>

### 服务端日志

<details><pre><code>
2024/11/04 04:57:50 [Debug] app/log: Logger started
2024/11/04 04:57:50 [Debug] app/proxyman/inbound: creating unix domain socket worker on @xhttp
2024/11/04 04:57:50 [Debug] app/proxyman/inbound: creating stream worker on 0.0.0.0:443
2024/11/04 04:57:50 [Debug] app/proxyman/inbound: creating stream worker on 0.0.0.0:8443
2024/11/04 04:57:50 [Info] transport/internet/splithttp: listening unix domain socket(for SH) on @xhttp
2024/11/04 04:57:50 [Info] transport/internet/tcp: listening TCP on 0.0.0.0:443
2024/11/04 04:57:50 [Info] transport/internet/tcp: listening TCP on 0.0.0.0:8443
2024/11/04 04:57:50 [Warning] core: Xray 24.10.31 started
2024/11/04 04:57:56 [Info] [897072903] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:56 [Info] [897072903] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:56 [Info] [897072903] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:56 [Info] [897072903] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:56 [Error] common/net: invalid IP format: []
2024/11/04 04:57:56 [Info] [1882259789] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:56 [Info] [1882259789] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:56 [Info] [1882259789] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:56 [Info] [1882259789] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:56 [Info] [711341622] proxy/vless/inbound: firstLen = 281
2024/11/04 04:57:56 [Info] [711341622] proxy/vless/inbound: received request for tcp:1.1.1.1:443
2024/11/04 04:57:56 [Info] [711341622] app/dispatcher: default route for tcp:1.1.1.1:443
2024/11/04 04:57:56 [Info] [711341622] transport/internet/tcp: dialing TCP to tcp:1.1.1.1:443
2024/11/04 04:57:56 [Debug] [711341622] transport/internet: dialing to tcp:1.1.1.1:443
2024/11/04 04:57:56 [Info] [711341622] proxy/freedom: connection opened to tcp:1.1.1.1:443, local endpoint 10.1.90.255:46880, remote endpoint 1.1.1.1:443
2024/11/04 04:57:56 [Info] [711341622] proxy: CopyRawConn readv
2024/11/04 04:57:56 [Info] [3916515550] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:56 [Info] [3916515550] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:56 [Info] [3916515550] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:56 [Info] [3916515550] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:56 [Error] common/net: invalid IP format: []
2024/11/04 04:57:56 [Info] [46717578] proxy/vless/inbound: firstLen = 281
2024/11/04 04:57:56 [Info] [46717578] proxy/vless/inbound: received request for tcp:1.1.1.1:443
2024/11/04 04:57:56 [Info] [46717578] app/dispatcher: default route for tcp:1.1.1.1:443
2024/11/04 04:57:56 [Info] [46717578] transport/internet/tcp: dialing TCP to tcp:1.1.1.1:443
2024/11/04 04:57:56 [Debug] [46717578] transport/internet: dialing to tcp:1.1.1.1:443
2024/11/04 04:57:56 [Info] [46717578] proxy/freedom: connection opened to tcp:1.1.1.1:443, local endpoint 10.1.90.255:46888, remote endpoint 1.1.1.1:443
2024/11/04 04:57:56 [Info] [46717578] proxy: CopyRawConn readv
2024/11/04 04:57:57 [Info] [802810446] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:57 [Info] [802810446] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:57 [Info] [802810446] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:57 [Info] [802810446] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:57 [Error] common/net: invalid IP format: []
2024/11/04 04:57:57 [Info] [1275141771] proxy/vless/inbound: firstLen = 556
2024/11/04 04:57:57 [Info] [1275141771] proxy/vless/inbound: received request for tcp:74.125.204.108:993
2024/11/04 04:57:57 [Info] [1275141771] app/dispatcher: default route for tcp:74.125.204.108:993
2024/11/04 04:57:57 [Info] [1275141771] transport/internet/tcp: dialing TCP to tcp:74.125.204.108:993
2024/11/04 04:57:57 [Debug] [1275141771] transport/internet: dialing to tcp:74.125.204.108:993
2024/11/04 04:57:57 [Info] [116720087] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:57 [Info] [116720087] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:57 [Info] [116720087] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:57 [Info] [116720087] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:57 [Error] common/net: invalid IP format: []
2024/11/04 04:57:57 [Info] [2709474395] proxy/vless/inbound: firstLen = 556
2024/11/04 04:57:57 [Info] [2709474395] proxy/vless/inbound: received request for tcp:74.125.204.108:993
2024/11/04 04:57:57 [Info] [2709474395] app/dispatcher: default route for tcp:74.125.204.108:993
2024/11/04 04:57:57 [Info] [2709474395] transport/internet/tcp: dialing TCP to tcp:74.125.204.108:993
2024/11/04 04:57:57 [Debug] [2709474395] transport/internet: dialing to tcp:74.125.204.108:993
2024/11/04 04:57:57 [Info] [2725242703] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:57 [Info] [2725242703] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:57 [Info] [2725242703] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:57 [Info] [2725242703] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:57 [Error] common/net: invalid IP format: []
2024/11/04 04:57:57 [Info] [3668283423] proxy/vless/inbound: firstLen = 192
2024/11/04 04:57:57 [Info] [3668283423] proxy/vless/inbound: received request for tcp:142.250.198.132:80
2024/11/04 04:57:57 [Info] [3668283423] app/dispatcher: default route for tcp:142.250.198.132:80
2024/11/04 04:57:57 [Info] [3668283423] transport/internet/tcp: dialing TCP to tcp:142.250.198.132:80
2024/11/04 04:57:57 [Debug] [3668283423] transport/internet: dialing to tcp:142.250.198.132:80
2024/11/04 04:57:57 [Info] [3668283423] proxy/freedom: connection opened to tcp:142.250.198.132:80, local endpoint 10.1.90.255:56042, remote endpoint 142.250.198.132:80
2024/11/04 04:57:57 [Info] [3668283423] proxy: CopyRawConn readv
2024/11/04 04:57:57 [Info] [1275141771] proxy/freedom: connection opened to tcp:74.125.204.108:993, local endpoint 10.1.90.255:38206, remote endpoint 74.125.204.108:993
2024/11/04 04:57:57 [Info] [1275141771] proxy: CopyRawConn readv
2024/11/04 04:57:57 [Info] [2709474395] proxy/freedom: connection opened to tcp:74.125.204.108:993, local endpoint 10.1.90.255:38214, remote endpoint 74.125.204.108:993
2024/11/04 04:57:57 [Info] [2709474395] proxy: CopyRawConn readv
2024/11/04 04:57:58 [Info] [782697407] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:58 [Info] [782697407] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:58 [Info] [782697407] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:58 [Info] [782697407] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:58 [Error] common/net: invalid IP format: []
2024/11/04 04:57:58 [Info] [609042049] proxy/vless/inbound: firstLen = 543
2024/11/04 04:57:58 [Info] [609042049] proxy/vless/inbound: received request for tcp:151.101.130.219:443
2024/11/04 04:57:58 [Info] [609042049] app/dispatcher: default route for tcp:151.101.130.219:443
2024/11/04 04:57:58 [Info] [609042049] transport/internet/tcp: dialing TCP to tcp:151.101.130.219:443
2024/11/04 04:57:58 [Debug] [609042049] transport/internet: dialing to tcp:151.101.130.219:443
2024/11/04 04:57:58 [Info] [3254854409] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:58 [Info] [3254854409] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:58 [Info] [3254854409] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:58 [Info] [3254854409] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:58 [Error] common/net: invalid IP format: []
2024/11/04 04:57:58 [Info] [609042049] proxy/freedom: connection opened to tcp:151.101.130.219:443, local endpoint 10.1.90.255:33794, remote endpoint 151.101.130.219:443
2024/11/04 04:57:58 [Info] [609042049] proxy: CopyRawConn readv
2024/11/04 04:57:58 [Info] [2945214519] proxy/vless/inbound: firstLen = 632
2024/11/04 04:57:58 [Info] [2945214519] proxy/vless/inbound: received request for tcp:23.35.148.25:443
2024/11/04 04:57:58 [Info] [2945214519] app/dispatcher: default route for tcp:23.35.148.25:443
2024/11/04 04:57:58 [Info] [2945214519] transport/internet/tcp: dialing TCP to tcp:23.35.148.25:443
2024/11/04 04:57:58 [Debug] [2945214519] transport/internet: dialing to tcp:23.35.148.25:443
2024/11/04 04:57:58 [Info] [2945214519] proxy/freedom: connection opened to tcp:23.35.148.25:443, local endpoint 10.1.90.255:45890, remote endpoint 23.35.148.25:443
2024/11/04 04:57:58 [Info] [2945214519] proxy: CopyRawConn readv
2024/11/04 04:57:59 [Info] [1186448835] proxy/vless/inbound: firstLen = 70
2024/11/04 04:57:59 [Info] [1186448835] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:57:59 [Info] [1186448835] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:57:59 [Info] [1186448835] proxy/vless/inbound: realAlpn = 
2024/11/04 04:57:59 [Error] common/net: invalid IP format: []
2024/11/04 04:57:59 [Info] [583665609] proxy/vless/inbound: firstLen = 547
2024/11/04 04:57:59 [Info] [583665609] proxy/vless/inbound: received request for tcp:104.17.148.22:443
2024/11/04 04:57:59 [Info] [583665609] app/dispatcher: default route for tcp:104.17.148.22:443
2024/11/04 04:57:59 [Info] [583665609] transport/internet/tcp: dialing TCP to tcp:104.17.148.22:443
2024/11/04 04:57:59 [Debug] [583665609] transport/internet: dialing to tcp:104.17.148.22:443
2024/11/04 04:57:59 [Info] [583665609] proxy/freedom: connection opened to tcp:104.17.148.22:443, local endpoint 10.1.90.255:55920, remote endpoint 104.17.148.22:443
2024/11/04 04:57:59 [Info] [583665609] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [475095849] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [475095849] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [475095849] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [475095849] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [3370964971] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:00 [Info] [3370964971] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:00 [Info] [3278533946] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [3278533946] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [3278533946] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [3278533946] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [1723146133] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:00 [Info] [1723146133] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:00 [Info] [1223805049] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [1223805049] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [1223805049] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [1223805049] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [4285828186] proxy/vless/inbound: firstLen = 53
2024/11/04 04:58:00 [Info] [4285828186] proxy/vless/inbound: received request for tcp:218.102.52.37:8080
2024/11/04 04:58:00 [Info] [4285828186] app/dispatcher: default route for tcp:218.102.52.37:8080
2024/11/04 04:58:00 [Info] [4285828186] transport/internet/tcp: dialing TCP to tcp:218.102.52.37:8080
2024/11/04 04:58:00 [Debug] [4285828186] transport/internet: dialing to tcp:218.102.52.37:8080
2024/11/04 04:58:00 [Info] [4057852979] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [4057852979] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [4057852979] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [4057852979] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Info] [1392155910] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [1392155910] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [1392155910] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [1392155910] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [1212595036] proxy/vless/inbound: firstLen = 65
2024/11/04 04:58:00 [Info] [1212595036] proxy/vless/inbound: received request for tcp:210.57.36.162:8080
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [1212595036] app/dispatcher: default route for tcp:210.57.36.162:8080
2024/11/04 04:58:00 [Info] [1212595036] transport/internet/tcp: dialing TCP to tcp:210.57.36.162:8080
2024/11/04 04:58:00 [Debug] [1212595036] transport/internet: dialing to tcp:210.57.36.162:8080
2024/11/04 04:58:00 [Info] [4285828186] proxy/freedom: connection opened to tcp:218.102.52.37:8080, local endpoint 10.1.90.255:51062, remote endpoint 218.102.52.37:8080
2024/11/04 04:58:00 [Info] [4285828186] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [738499461] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [738499461] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [738499461] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [738499461] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Info] [437149969] proxy/vless/inbound: firstLen = 53
2024/11/04 04:58:00 [Info] [437149969] proxy/vless/inbound: received request for tcp:156.0.200.220:8080
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [437149969] app/dispatcher: default route for tcp:156.0.200.220:8080
2024/11/04 04:58:00 [Info] [437149969] transport/internet/tcp: dialing TCP to tcp:156.0.200.220:8080
2024/11/04 04:58:00 [Debug] [437149969] transport/internet: dialing to tcp:156.0.200.220:8080
2024/11/04 04:58:00 [Info] [584365644] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [584365644] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [584365644] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [584365644] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Info] [3376912393] proxy/vless/inbound: firstLen = 45
2024/11/04 04:58:00 [Info] [3376912393] proxy/vless/inbound: received request for tcp:203.210.4.91:8080
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [3376912393] app/dispatcher: default route for tcp:203.210.4.91:8080
2024/11/04 04:58:00 [Info] [3376912393] transport/internet/tcp: dialing TCP to tcp:203.210.4.91:8080
2024/11/04 04:58:00 [Debug] [3376912393] transport/internet: dialing to tcp:203.210.4.91:8080
2024/11/04 04:58:00 [Info] [437149969] proxy/freedom: connection opened to tcp:156.0.200.220:8080, local endpoint 10.1.90.255:51686, remote endpoint 156.0.200.220:8080
2024/11/04 04:58:00 [Info] [437149969] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [2105537779] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [2105537779] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [2105537779] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [2105537779] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Info] [3304698411] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [3304698411] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [3304698411] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [3304698411] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Info] [2513188248] proxy/vless/inbound: firstLen = 49
2024/11/04 04:58:00 [Info] [2513188248] proxy/vless/inbound: received request for tcp:210.0.250.30:8080
2024/11/04 04:58:00 [Info] [2513188248] app/dispatcher: default route for tcp:210.0.250.30:8080
2024/11/04 04:58:00 [Info] [2513188248] transport/internet/tcp: dialing TCP to tcp:210.0.250.30:8080
2024/11/04 04:58:00 [Debug] [2513188248] transport/internet: dialing to tcp:210.0.250.30:8080
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [3376912393] proxy/freedom: connection opened to tcp:203.210.4.91:8080, local endpoint 10.1.90.255:36960, remote endpoint 203.210.4.91:8080
2024/11/04 04:58:00 [Info] [3376912393] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [2615759287] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [2615759287] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [2615759287] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [2615759287] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [2513188248] proxy/freedom: connection opened to tcp:210.0.250.30:8080, local endpoint 10.1.90.255:49776, remote endpoint 210.0.250.30:8080
2024/11/04 04:58:00 [Info] [2513188248] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [3909725177] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [3909725177] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [3909725177] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [3909725177] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [2947057576] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:00 [Info] [2947057576] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:00 [Info] [2947057576] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:00 [Info] [2947057576] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:00 [Error] common/net: invalid IP format: []
2024/11/04 04:58:00 [Info] [4234816597] proxy/vless/inbound: firstLen = 69
2024/11/04 04:58:00 [Info] [4234816597] proxy/vless/inbound: received request for tcp:203.210.2.92:8080
2024/11/04 04:58:00 [Info] [4234816597] app/dispatcher: default route for tcp:203.210.2.92:8080
2024/11/04 04:58:00 [Info] [4234816597] transport/internet/tcp: dialing TCP to tcp:203.210.2.92:8080
2024/11/04 04:58:00 [Debug] [4234816597] transport/internet: dialing to tcp:203.210.2.92:8080
2024/11/04 04:58:00 [Info] [3203673213] proxy/vless/inbound: firstLen = 60
2024/11/04 04:58:00 [Info] [3203673213] proxy/vless/inbound: received request for tcp:203.78.41.10:8080
2024/11/04 04:58:00 [Info] [795902724] proxy/vless/inbound: firstLen = 48
2024/11/04 04:58:00 [Info] [795902724] proxy/vless/inbound: received request for tcp:61.244.99.206:8080
2024/11/04 04:58:00 [Info] [475951970] proxy/vless/inbound: firstLen = 62
2024/11/04 04:58:00 [Info] [475951970] proxy/vless/inbound: received request for tcp:43.239.139.213:8080
2024/11/04 04:58:00 [Info] [1134830542] proxy/vless/inbound: firstLen = 52
2024/11/04 04:58:00 [Info] [1134830542] proxy/vless/inbound: received request for tcp:218.102.20.42:8080
2024/11/04 04:58:00 [Info] [3203673213] app/dispatcher: default route for tcp:203.78.41.10:8080
2024/11/04 04:58:00 [Info] [3203673213] transport/internet/tcp: dialing TCP to tcp:203.78.41.10:8080
2024/11/04 04:58:00 [Debug] [3203673213] transport/internet: dialing to tcp:203.78.41.10:8080
2024/11/04 04:58:00 [Info] [795902724] app/dispatcher: default route for tcp:61.244.99.206:8080
2024/11/04 04:58:00 [Info] [795902724] transport/internet/tcp: dialing TCP to tcp:61.244.99.206:8080
2024/11/04 04:58:00 [Debug] [795902724] transport/internet: dialing to tcp:61.244.99.206:8080
2024/11/04 04:58:00 [Info] [475951970] app/dispatcher: default route for tcp:43.239.139.213:8080
2024/11/04 04:58:00 [Info] [475951970] transport/internet/tcp: dialing TCP to tcp:43.239.139.213:8080
2024/11/04 04:58:00 [Debug] [475951970] transport/internet: dialing to tcp:43.239.139.213:8080
2024/11/04 04:58:00 [Info] [1134830542] app/dispatcher: default route for tcp:218.102.20.42:8080
2024/11/04 04:58:00 [Info] [1134830542] transport/internet/tcp: dialing TCP to tcp:218.102.20.42:8080
2024/11/04 04:58:00 [Debug] [1134830542] transport/internet: dialing to tcp:218.102.20.42:8080
2024/11/04 04:58:00 [Info] [1212595036] proxy/freedom: connection opened to tcp:210.57.36.162:8080, local endpoint 10.1.90.255:48744, remote endpoint 210.57.36.162:8080
2024/11/04 04:58:00 [Info] [1212595036] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [4234816597] proxy/freedom: connection opened to tcp:203.210.2.92:8080, local endpoint 10.1.90.255:52262, remote endpoint 203.210.2.92:8080
2024/11/04 04:58:00 [Info] [4234816597] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [475951970] proxy/freedom: connection opened to tcp:43.239.139.213:8080, local endpoint 10.1.90.255:59328, remote endpoint 43.239.139.213:8080
2024/11/04 04:58:00 [Info] [475951970] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [3203673213] proxy/freedom: connection opened to tcp:203.78.41.10:8080, local endpoint 10.1.90.255:52538, remote endpoint 203.78.41.10:8080
2024/11/04 04:58:00 [Info] [3203673213] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [1134830542] proxy/freedom: connection opened to tcp:218.102.20.42:8080, local endpoint 10.1.90.255:40370, remote endpoint 218.102.20.42:8080
2024/11/04 04:58:00 [Info] [1134830542] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [795902724] proxy/freedom: connection opened to tcp:61.244.99.206:8080, local endpoint 10.1.90.255:47578, remote endpoint 61.244.99.206:8080
2024/11/04 04:58:00 [Info] [795902724] proxy: CopyRawConn readv
2024/11/04 04:58:00 [Info] [583665609] app/proxyman/inbound: connection ends > proxy/vless/inbound: connection ends > context canceled
2024/11/04 04:58:01 [Info] [3583447077] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:01 [Info] [3583447077] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:01 [Info] [3583447077] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:01 [Info] [3583447077] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:01 [Error] common/net: invalid IP format: []
2024/11/04 04:58:01 [Info] [1854640074] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:01 [Info] [1854640074] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:01 [Info] [3650894846] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:01 [Info] [3650894846] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:01 [Info] [3650894846] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:01 [Info] [3650894846] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:01 [Info] [978266105] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:01 [Info] [978266105] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:01 [Info] [978266105] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:01 [Info] [978266105] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:01 [Error] common/net: invalid IP format: []
2024/11/04 04:58:01 [Info] [426552193] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:01 [Info] [426552193] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:01 [Error] common/net: invalid IP format: []
2024/11/04 04:58:01 [Info] [2103516237] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:01 [Info] [2103516237] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:01 [Info] [3112864690] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:01 [Info] [3112864690] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:01 [Info] [3112864690] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:01 [Info] [3112864690] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:01 [Error] common/net: invalid IP format: []
2024/11/04 04:58:01 [Info] [319658106] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:01 [Info] [319658106] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:01 [Info] [2940963048] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:01 [Info] [2940963048] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:01 [Info] [2940963048] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:01 [Info] [2940963048] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:01 [Error] common/net: invalid IP format: []
2024/11/04 04:58:01 [Info] [4132824136] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:01 [Info] [4132824136] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:02 [Info] [3052200098] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:02 [Info] [3052200098] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:02 [Info] [3052200098] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:02 [Info] [3052200098] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:02 [Error] common/net: invalid IP format: []
2024/11/04 04:58:02 [Info] [1960971638] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:02 [Info] [1960971638] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:02 [Info] [1907158720] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:02 [Info] [1907158720] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:02 [Info] [1907158720] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:02 [Info] [1907158720] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:02 [Info] [1270709169] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:02 [Info] [1270709169] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:02 [Info] [1270709169] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:02 [Info] [1270709169] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:02 [Error] common/net: invalid IP format: []
2024/11/04 04:58:02 [Info] [3738378217] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:02 [Info] [3738378217] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:02 [Error] common/net: invalid IP format: []
2024/11/04 04:58:02 [Info] [1284638790] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:02 [Info] [1284638790] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:03 [Info] [3376912393] app/proxyman/inbound: connection ends > proxy/vless/inbound: connection ends > context canceled
2024/11/04 04:58:03 [Info] [437149969] app/proxyman/inbound: connection ends > proxy/vless/inbound: connection ends > context canceled
2024/11/04 04:58:03 [Info] [2513188248] app/proxyman/inbound: connection ends > proxy/vless/inbound: connection ends > context canceled
2024/11/04 04:58:03 [Info] [303041411] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:03 [Info] [303041411] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:03 [Info] [303041411] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:03 [Info] [303041411] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:03 [Error] common/net: invalid IP format: []
2024/11/04 04:58:03 [Info] [1586127422] proxy/vless/inbound: firstLen = 543
2024/11/04 04:58:03 [Info] [1586127422] proxy/vless/inbound: received request for tcp:151.101.194.219:443
2024/11/04 04:58:03 [Info] [1586127422] app/dispatcher: default route for tcp:151.101.194.219:443
2024/11/04 04:58:03 [Info] [1586127422] transport/internet/tcp: dialing TCP to tcp:151.101.194.219:443
2024/11/04 04:58:03 [Debug] [1586127422] transport/internet: dialing to tcp:151.101.194.219:443
2024/11/04 04:58:03 [Info] [1586127422] proxy/freedom: connection opened to tcp:151.101.194.219:443, local endpoint 10.1.90.255:40412, remote endpoint 151.101.194.219:443
2024/11/04 04:58:03 [Info] [1586127422] proxy: CopyRawConn readv
2024/11/04 04:58:04 [Info] [2399022851] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:04 [Info] [2399022851] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:04 [Info] [2399022851] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:04 [Info] [2399022851] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:04 [Error] common/net: invalid IP format: []
2024/11/04 04:58:04 [Info] [2383979813] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:04 [Info] [2383979813] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:04 [Info] [4251140067] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:04 [Info] [4251140067] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:04 [Info] [4251140067] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:04 [Info] [4251140067] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:04 [Error] common/net: invalid IP format: []
2024/11/04 04:58:04 [Info] [2099595631] proxy/vless/inbound: firstLen = 19
2024/11/04 04:58:04 [Info] [2099595631] proxy/vless/inbound: received request for tcp:v1.mux.cool:0
2024/11/04 04:58:04 [Info] [1850151499] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:04 [Info] [1850151499] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:04 [Info] [1850151499] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:04 [Info] [1850151499] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:04 [Error] common/net: invalid IP format: []
2024/11/04 04:58:04 [Info] [1861991326] proxy/vless/inbound: firstLen = 66
2024/11/04 04:58:04 [Info] [1861991326] proxy/vless/inbound: received request for tcp:203.210.2.92:8080
2024/11/04 04:58:04 [Info] [1861991326] app/dispatcher: default route for tcp:203.210.2.92:8080
2024/11/04 04:58:04 [Info] [1861991326] transport/internet/tcp: dialing TCP to tcp:203.210.2.92:8080
2024/11/04 04:58:04 [Debug] [1861991326] transport/internet: dialing to tcp:203.210.2.92:8080
2024/11/04 04:58:04 [Info] [1861991326] proxy/freedom: connection opened to tcp:203.210.2.92:8080, local endpoint 10.1.90.255:52268, remote endpoint 203.210.2.92:8080
2024/11/04 04:58:04 [Info] [1861991326] proxy: CopyRawConn readv
2024/11/04 04:58:09 [Info] [348810481] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:09 [Info] [348810481] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:09 [Info] [348810481] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:09 [Info] [348810481] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:09 [Error] common/net: invalid IP format: []
2024/11/04 04:58:09 [Info] [3129775357] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:09 [Info] [3129775357] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:09 [Info] [3129775357] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:09 [Info] [3129775357] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:09 [Error] common/net: invalid IP format: []
2024/11/04 04:58:09 [Info] [3330055187] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:09 [Info] [3330055187] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:09 [Info] [3808350275] proxy/vless/inbound: firstLen = 70
2024/11/04 04:58:09 [Info] [3808350275] proxy/vless/inbound: fallback starts > proxy/vless/encoding: invalid request version
2024/11/04 04:58:09 [Info] [3808350275] proxy/vless/inbound: realName = www.whitehouse.gov
2024/11/04 04:58:09 [Info] [3808350275] proxy/vless/inbound: realAlpn = 
2024/11/04 04:58:09 [Error] common/net: invalid IP format: []
2024/11/04 04:58:09 [Info] [1541638761] proxy/vless/inbound: firstLen = 282
2024/11/04 04:58:09 [Info] [1541638761] proxy/vless/inbound: received request for tcp:v1.mux.cool:0
2024/11/04 04:58:09 [Info] [1132631242] proxy/vless/inbound: firstLen = 0
2024/11/04 04:58:09 [Info] [1132631242] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2024/11/04 04:58:09 [Info] [1541638761] common/mux: received request for udp:156.0.200.220:8080
2024/11/04 04:58:09 [Info] [1541638761] common/mux: XUDP new [21 15 135 118 44 127 218 22]
2024/11/04 04:58:09 [Info] [1541638761] app/dispatcher: default route for udp:156.0.200.220:8080
2024/11/04 04:58:09 [Debug] [1541638761] transport/internet: dialing to udp:156.0.200.220:8080
2024/11/04 04:58:09 [Info] [1541638761] proxy/freedom: connection opened to udp:156.0.200.220:8080, local endpoint [::]:49394, remote endpoint 156.0.200.220:8080
</code></pre></details>

--- a/app/proxyman/inbound/worker.go
+++ b/app/proxyman/inbound/worker.go
@@ -464,7 +464,8 @@ func (w *dsWorker) callback(conn stat.Connection) {
 		}
 	}
 	ctx = session.ContextWithInbound(ctx, &session.Inbound{
-		Source:  net.DestinationFromAddr(conn.RemoteAddr()),
+		// Unix have no source addr, so we use gateway as source for log.
+		Source:  net.UnixDestination(w.address),
 		Gateway: net.UnixDestination(w.address),
 		Tag:     w.tag,
 		Conn:    conn,

--- a/common/net/destination.go
+++ b/common/net/destination.go
@@ -89,10 +89,12 @@ func UnixDestination(address Address) Destination {
 // NetAddr returns the network address in this Destination in string form.
 func (d Destination) NetAddr() string {
 	addr := ""
-	if d.Network == Network_TCP || d.Network == Network_UDP {
-		addr = d.Address.String() + ":" + d.Port.String()
-	} else if d.Network == Network_UNIX {
-		addr = d.Address.String()
+	if d.Address != nil {
+		if d.Network == Network_TCP || d.Network == Network_UDP {
+			addr = d.Address.String() + ":" + d.Port.String()
+		} else if d.Network == Network_UNIX {
+			addr = d.Address.String()
+		}
 	}
 	return addr
 }

